test/examples_test.go

```diff
@@ -0,0 +1,399 @@
+)
+
+func validateObject(obj runtime.Object) (errors field.ErrorList) {
+	switch t := obj.(type) {
+	case *api.ReplicationController:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidateReplicationController(t)
+	case *api.ReplicationControllerList:
+		for i := range t.Items {
+			errors = append(errors, validateObject(&t.Items[i])...)
+		}
+	case *api.Service:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidateService(t)
+	case *api.ServiceList:
+		for i := range t.Items {
+			errors = append(errors, validateObject(&t.Items[i])...)
+		}
+	case *api.Pod:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidatePod(t)
+	case *api.PodList:
+		for i := range t.Items {
+			errors = append(errors, validateObject(&t.Items[i])...)
+		}
+	case *api.PersistentVolume:
+		errors = validation.ValidatePersistentVolume(t)
+	case *api.PersistentVolumeClaim:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidatePersistentVolumeClaim(t)
+	case *api.PodTemplate:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidatePodTemplate(t)
+	case *api.Endpoints:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidateEndpoints(t)
+	case *api.Namespace:
+		errors = validation.ValidateNamespace(t)
+	case *api.Secret:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidateSecret(t)
+	case *api.LimitRange:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidateLimitRange(t)
+	case *api.ResourceQuota:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = validation.ValidateResourceQuota(t)
+	case *extensions.Deployment:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = expvalidation.ValidateDeployment(t)
+	case *batch.Job:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		// Job needs generateSelector called before validation, and job.Validate does this.
+		// See: https://github.com/kubernetes/kubernetes/issues/20951#issuecomment-187787040
+		t.ObjectMeta.UID = types.UID("fakeuid")
+		errors = job.Strategy.Validate(nil, t)
+	case *extensions.Ingress:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = expvalidation.ValidateIngress(t)
+	case *extensions.DaemonSet:
+		if t.Namespace == "" {
+			t.Namespace = api.NamespaceDefault
+		}
+		errors = expvalidation.ValidateDaemonSet(t)
+	default:
+		errors = field.ErrorList{}
+		errors = append(errors, field.InternalError(field.NewPath(""), fmt.Errorf("no validation defined for %#v", obj)))
+	}
+	return errors
+}
+
+func walkJSONFiles(inDir string, fn func(name, path string, data []byte)) error {
```

Comment:
Could you please add a comment as to what this function does? Also, the name of the function isn't indicative of the fact that we are looking at both YAML and JSON files.